client/tests/reload.rs

```diff
@@ -0,0 +1,20 @@
+#![allow(clippy::restriction)]
+use iroha_logger::Level::*;
+use test_network::Peer as TestPeer;
+
+#[test]
+fn reload_log_level() {
+    // The underscored variables must not be dropped until end of closure.
```

Comment:
Generally the comment about not dropping the runtime I think can be moved into description of start test with runtime function.